Commands/GenerateWebpImages.php

```diff
@@ -28,36 +61,155 @@ protected function configure()
+
+        $progressBar = new ProgressBar($this->output, $total);
+        $progressBar->setRedrawFrequency(10);
+        $progressBar->start();
+
+        /* @var Media $media */
+        foreach ($paginator->getIterator() as $media) {
             try {
-                $im = imagecreatefromstring($this->container->get('shopware_media.media_service')->read($item['path']));
-                if ($im === false) {
-                    throw new \Exception('Could not load image');
-                }
-                imagepalettetotruecolor($im);
-                $content = current($runnableEncoders)->encode($im, $this->container->get('frosh_webp.config')['webPQuality']);
-                imagedestroy($im);
-                $this->container->get('shopware_media.media_service')->write($webpPath, $content);
+                $this->createWebPImage($media);
             } catch (\Exception $e) {
-                $output->writeln($item['path'] . ' => ' . $e->getMessage());
+                $this->errors[] = $e->getMessage();
             } catch (\Throwable $e) {
-                $output->writeln($item['path'] . ' => ' . $e->getMessage());
+                $this->errors[] = $e->getMessage();
             }
-            $progress->advance();
+
+            $progressBar->advance();
+        }
+
+        $progressBar->finish();
+
+        // Force newline when processing the next album
+        $this->output->writeln('');
+    }
+
+    /**
+     * @param Media $media
+     *
+     * @throws \Exception
+     */
+    private function createWebPImage(Media $media)
+    {
+        $config = $this->container->get('frosh_webp.config');
+        $mediaService = $this->container->get('shopware_media.media_service');
+
+        $extension = strtolower($media->getExtension());
+
+        if (!in_array($extension, self::CAN_HANDLE_EXTENSIONS)) {
+            return;
+        }
+
+        $webpPath = str_replace($media->getExtension(), 'webp', $media->getPath());
+
+        if (!$this->imageExists($media)) {
+            throw new \Exception(sprintf('Base image file "%s" does not exist', $media->getPath()));
+        }
+
+        $image = imagecreatefromstring($mediaService->read($media->getPath()));
+
+        if ($image === false) {
+            throw new \Exception(sprintf('Could not load base image file "%s"', $media->getPath()));
+        }
+
+        imagepalettetotruecolor($image);
+        $content = current($this->runnableEncoders)->encode($image, $config['webPQuality']);
+        imagedestroy($image);
+
+        $mediaService->write($webpPath, $content);
+    }
+
+    /**
+     * @param Media $media
+     *
+     * @throws \Exception
+     *
+     * @return bool
+     */
+    private function imageExists(Media $media)
+    {
+        $mediaService = $this->container->get('shopware_media.media_service');
+        $projectDir = $this->container->getParameter('shopware.app.rootdir');
+
+        return $mediaService->has($projectDir . $media->getPath());
```

Comment:
Why preprending $projectDir?

Reply:
🤔 copied from Shopware thumbnail generation command.

Reply:
$media->getPath should be enought